application/include/mmul.hpp

```diff
@@ -51,8 +50,18 @@ void mmul(const Executor ex, const MatrixXd& a, const MatrixXd& b,
 }
 
 template <typename Executor,
-          typename executor::instance_of_base<cuda_executor, Executor> = 0>
+          typename executor::InstanceOfAny<Executor, cuda_executor> = 0>
 void mmul(const Executor& ex, const MatrixXd& a, const MatrixXd& b,
           MatrixXd& c) {
   mmul_gpu(ex, a, b, c);
 }
+
+void mmul(const MatrixXd& a, const MatrixXd& b, MatrixXd& c,
+          bool custom_priority) {
+  auto mul = [&](auto& exec) { mmul(exec, a, b, c); };
+
+  if (custom_priority)
+    enable_exec_with_priority(mul, inline_executor<>{}, omp_executor<>{});
+  else
+    enable_exec_on_desc_priority(mul, inline_executor<>{}, omp_executor<>{});
```

Comment:
Okay so the main purpose the user needs to specify the executor is that there is no way for `enable_exec_on_desc_priority` to know whether the function you passed has an overload for the executor in the default sequence.

Example:
If in `FunctorFilter` there exits an overload for `applyFilter` for OMP and Inline. 
If we don't pass the set of executors to `enable_exec_on_desc_priority` then it will try out SSE executor too since it is specified in the default set of executors by the library, this leads to a compile error.